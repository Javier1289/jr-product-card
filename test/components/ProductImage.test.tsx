import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/Products';

describe('ProductImage', () => {

     test( 'Debe mostrar el componente correctamente', () => {

        const wrapper = renderer.create(
            <ProductImage img={ product2.img } />
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();


     })


     test( 'Debe mostrar el componente con la imagen', () => {

        const wrapper = renderer.create(
            <ProductCard
                product={ product2 }
            >
                {
                    () => (
                        <ProductImage/>
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();



     })

})