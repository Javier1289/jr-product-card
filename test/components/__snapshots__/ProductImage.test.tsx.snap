// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar el componente con la imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;

exports[`ProductImage Debe mostrar el componente correctamente 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="./coffe-mug.png"
/>
`;
